crates/sui-core/src/authority.rs

```diff
@@ -3006,12 +3015,20 @@ impl AuthorityState {
     ) -> Option<ObjectRef> {
         let cur_object = match self.get_object(&id).await {
             Ok(Some(cur_object)) => cur_object,
-
             Ok(None) => {
-                error!("No framework package at {id}");
-                return None;
+                // creating a new framework package--nothing to check
+                return Some(
+                    Object::new_system_package(
+                        modules,
+                        SequenceNumber::new(),
+                        dependencies,
+                        // TODO: what should this be? looking at the current call sites, we're not necessarily calling compare_system_package from inside
```

Comment:
Particularly seeking advice on this question

Reply:
Genesis is fine here, we only use it to calculate an object ref that we can use for all validators to commit to the same bytes in the update.  For other packages we use the digest for the transaction that created the version that is currently in use so that if the bytes are the same as before, then the digest will also be the same and we can do the equality check on digests.  For new packages that isn't a factor.

Reply:
Thanks! I'll update the comment to reflect this.